examples/common/mod.rs

```diff
@@ -6,9 +6,9 @@ use futures::prelude::*;
 pub fn authenticate_client<'a>(
     client: &'a mut dkregistry::v2::Client,
     login_scope: &'a str,
-) -> impl futures::future::Future<Item = &'a dkregistry::v2::Client, Error = dkregistry::errors::Error>
+) -> impl futures::future::Future<Item = dkregistry::v2::Client, Error = dkregistry::errors::Error> + 'a
```

Comment:
I am not sure if I'm missing something, but it feels like this function could just take ownership of the input client and get rid of the lifetime. It is anyway cloning internally and returning a fully owned client on success.

Reply:
Also, non-namespaced `Future` should be already in scope here as there is a prelude-use at the top-level.